### ISSUE TITLE
Axios post request network error on android  

### ISSUE THREAD

**Issue by Zakisb** [axios/axios#4823]
#### Describe the bug
0

So i'm posting a formdata object with axios to a node.js server. On iOS everything works perfectly, the data get posted and the image uploaded. But on android i'm getting this error. PLEASE NOTE THAT I DID READ THE PREVIOUS ISSUES AND TRIED THE SOLUTION. NOTHING WORKED !

#### To Reproduce
Code snippet to reproduce, ideally that will work by pasting into something like https://npm.runkit.com/axios, a hosted solution, or a repository that illustrates the issue. **If your problem is not reproducible, please file under Support or Usage Question**

```js
const handleSubmit = async (listing, { resetForm }) => {
    
    const data = new FormData();
listing.images.forEach((image, index) =>
      data.append("images", {
        name: `product${Math.floor(Math.random() * 1000)}.`,
        uri: image,
      })
    );

    const res = await axios
      .post("http://192.168.43.8:5000/products/addProduct", data, {
        headers: {
          "Content-Type": "multipart/form-data",
        },
      //tried adding this but didn't work out
        transformRequest: (data) => data,
      })
      .then(function (response) {
        console.log(response);
      })
      .catch(function (error) {
        console.log(error);
        // handle error
      });
  }


}
```
Please note that on iOS it works without a problem. here's a screenshot of the error when i used the react native debugger
![Screen Shot 2022-06-29 at 5 26 14 PM](https://user-images.githubusercontent.com/28612240/176488973-c94d80b1-ee08-4988-acae-cfeae9b1da6a.png)

#### Expected behavior
FORMDATA OBJECT POSTED TO NODE.JS server 

#### Environment
 - Axios Version 0.27.2

 - Adapter : HTTP
 - Browser : Chrome
 - Browser :  Version 103.0.5060.53 (Official Build) (arm64)
 - Node.js  : v14.17.5
 - OS: macOS big sur version 11.4
 - expo: "~45.0.0"
 - "react-native": "0.68.2",

#### Additional context/Screenshots
Add any other context about the problem here. If applicable, add screenshots to help explain.

**Comment by hemandpvkd** [comment]
Downgrade axios version to 0.24.0

**Comment by federicogomezcactus** [comment]
> Downgrade axios version to 0.24.0

Tried this but it didn't work, I have the same issue (I was on 0.27.2 before if i'm correct)

**Comment by isaque2304** [comment]
I have the same problem, but mine only gives an error when I use android 5.1.1

**Comment by habrahamyanmabbly** [comment]
same issue

**Comment by Quintis1212** [comment]
Is there any fix ?

**Comment by Quintis1212** [comment]
@Zakisb  Hi . Could you please delete node_modules folder and then run  npm install ?

**Comment by babar-ali-memon** [comment]
@Quintis1212 I'm facing same issue also tried deleting node_modules and reinstalling but no luck

**Comment by mlazari** [comment]
Ok, so after a few hours of debugging, I finally found what's wrong - it's the `Content-Type` request header.
If you don't set it explicitly, then the default value is set here to `application/x-www-form-urlencoded` -> https://github.com/axios/axios/blob/v0.27.2/lib/defaults/index.js#L10 and that apparently makes the request fail on Android, but not on iOS.
Setting the `'Content-Type': 'multipart/form-data'` explicitly fixes the error.
It was working in 0.26.1 because this code was removing the `Content-Type` header: https://github.com/axios/axios/blob/v0.26.1/lib/adapters/xhr.js#L32

**Comment by singhgovind770** [comment]
Getting Same issue, Is there any fix ?

**Comment by Mkay4real** [comment]
I'm also facing this issue

**Comment by Umion** [comment]
This solution worked for me
If you are using an emulator, change the ip address
from 
`http://localhost:5000/api` to `http://10.0.2.2:5000/api`

**Comment by alexandrius** [comment]
> @Mkay4real @singhgovind770 Did you try to set `'Content-Type': 'multipart/form-data'` in `headers` when you make the request?

Can confirm setting 'Content-Type': 'multipart/form-data' makes it work again

**Comment by liylmn** [comment]
When Android uploads videos, the type of formData is set to 'multipart/form data', and it can run! Previously set 'video'...

**Comment by tsafadi** [comment]
This is causing some serious pain in the development process. This should be fixed asap.

**Comment by cremirdevio** [comment]
Like its been mentioned above;
Setting the `'Content-Type': 'multipart/form-data'` explicitly fixes the error.

If it doesn't, make sure the type on your Blob is a valid MimeType.
In my case, I was passing `type: 'image'`, in iOS, it worked fine, but on android, it didn't work until I updated it to `type: 'image/jpeg'` accordingly.

**Comment by sujathasperi2022** [comment]
I am also getting the same error in android device & simulator
Setting `'Content-Type': 'multipart/form-data'` didin't help.
For me the error happens only for post request - with maultipartform data
Other GET apis are working correctly. 
Funny thing is all api works correctly in iOS

**Comment by habrahamyanmabbly** [comment]
Here is my example for expo applications 




```
import * as FileSystem from 'expo-file-system';



return await FileSystem.uploadAsync(
				`endpoint`,
				uri,
				{
					fieldName: "Video Name",
					mimeType: 'video',
					httpMethod: 'POST',
					uploadType: FileSystem.FileSystemUploadType.MULTIPART,
					headers: {
						Accept: 'application/json',
						'Content-Type': 'multipart/form-data',
						Authorization: `Basic ${base64}`,
					},
				},
			);
```

**Comment by fouad1994** [comment]
> Downgrade axios version to 0.24.0

Thanks, it works for me

**Comment by cremirdevio** [comment]
> Here is my example for expo applications 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> ```
> 
> import * as FileSystem from 'expo-file-system';
> 
> 
> 
>          try {
> 
> 		return await FileSystem.uploadAsync(
> 
> 				`endpoint_url`,
> 
> 				video_uri
> 
> 				{
> 
> 					fieldName: "Video Name",
> 
> 					mimeType: 'video',
> 
> 					httpMethod: 'POST',
> 
> 					uploadType: FileSystem.FileSystemUploadType.MULTIPART,
> 
> 					headers: {
> 
> 						Accept: 'application/json',
> 
> 						'Content-Type': 'multipart/form-data',
> 
> 						Authorization: `Basic ${Token if needed}`,
> 
> 					},
> 
> 				},
> 
> 			);
> 
> 		} catch (error) {
> 
> 			console.log(error)
> 
> 		}
> 
> ```

Video is not a valid mimeType. Try changing your mimeType to the exact mimeType like 'video/mp4', 'video/ogg' etc. 

In my case, for image upload, changing the type from 'image' to 'image/jpeg' solved the issue for me.

**Comment by MoTechStore** [comment]
Downgrade axios version to 0.24.0 also worked for me.
With     'Content-Type': 'application/json',
                 'Accept': 'application/json',

**Comment by valentingnt** [comment]
Hello guys !
I've been struggling with this issue too, and tried every pieces of solution I could find in your answers. I'm on RN 0.67.5, with axios v0.24.0. Did somebody find any other solution than the ones mentioned above ?

**Comment by jan-karnik** [comment]
> Like its been mentioned above; Setting the `'Content-Type': 'multipart/form-data'` explicitly fixes the error.
> 
> If it doesn't, make sure the type on your Blob is a valid MimeType. In my case, I was passing `type: 'image'`, in iOS, it worked fine, but on android, it didn't work until I updated it to `type: 'image/jpeg'` accordingly.

This worked for me, thanks! I was also passing `type: 'image'` to the Form data, changed it to `image/jpeg` for Android and it works!

**Comment by ConnalATU** [comment]
I'm having this problem, I changed the content-type and the post now works, however when trying to add data to a database it just adds null. It works perfect on react.js, does anyone please have a solution?

**Comment by ConnalATU** [comment]
just used fetch over axios working now

**Comment by vassbg** [comment]
@mlazari Thank you!!!
'Content-Type': 'multipart/form-data' fixed it for me!

**Comment by molo0** [comment]
Hey guys, I had the same problem. Axios works well at iOS and some android devices. In my case, It was my server's protocol setting.
Check here.
[React Native fetch error comment](https://github.com/facebook/react-native/issues/37382#issuecomment-1550793801)

On the other way, I found server's ssl certification file can cause this error too.

**Comment by Josh2941** [comment]
I was getting the same error. Spent several days and a lot of time upgrading packages and trying all the solutions from issues and stackoverflow. Finally this was the mistake which I don't think is mentioned which even in the docs.

If we have a `FormData` object

```
formData.append("name", "abc")
formData.append("state", undefined)
formData.append("address", {
    addressLine1: "",
    addressLine2: ""
});

```
It will throw the same error since the `"state"` is `undefined` and `"address"` is an object. But `FormData` accepts only string or File or Blob. So this was causing the issue for me. Once I replaced `undefined` with `""` and `"address"` with `JSON.stringify(addressObj)` it started working.

Reference -> https://stackoverflow.com/a/72899049/5918233

**Comment by JeffreyVanelderenACA** [comment]
For me, it was **not** related to form data or anything **frontend** related! On Android only, I could not reach our production server. Axios gave an `ERR_NETWORK` error without more info. This was working on iOS! And dev/staging URLs were working perfectly fine on the same Android device.

I asked a friend to execute the same Axios request as in my app in his own Android app and he got the same `ERR_NETWORK` error so this meant it was not an issue specific to my Android app...
Then I created the fetch version request of my Axios request. example:
```
await fetch('URL', {
  method: 'GET',
  headers: {
    'Content-Type': 'application/json',
  },
})
  .then(async response =>
    console.log(
      'TEST SUCCESS',
      JSON.stringify(response),
      await response.json(),
    ),
  )
  .catch(response => console.log('TEST ERROR', JSON.stringify(response)));
```

And executed this via Chrome dev tools, I then saw a CORS related error (see screenshot below) and passed this to the Backend developers. They changed the CORS policy and voila, it was fixed...

<img width="754" alt="Screenshot 2025-01-09 at 14 04 54" src="https://github.com/user-attachments/assets/88265858-5ffb-4e26-84ca-26a741505617" />
